kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KopEventManager.java

```diff
@@ -0,0 +1,291 @@
+
+    public KopEventManager(GroupCoordinator coordinator,
+                           AdminManager adminManager,
+                           MetadataStore metadataStore) {
+        this.coordinator = coordinator;
+        this.adminManager = adminManager;
+        this.deletionTopicsHandler = new DeletionTopicsHandler(this);
+        this.brokersChangeHandler = new BrokersChangeHandler(this);
+        this.metadataStore = metadataStore;
+    }
+
+    public void start() {
+        registerChildChangeHandler();
+        thread.start();
+    }
+
+    public void close() {
+        try {
+            thread.shutdown();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            log.error("Interrupted at shutting down {}", kopEventThreadName);
+        }
+
+    }
+
+
+    public void put(KopEvent event) {
+        try {
+            putLock.lock();
+            queue.put(event);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            log.error("Error put event {} to coordinator event queue {}", event, e);
+        } finally {
+            putLock.unlock();
+        }
+    }
+
+    public void clearAndPut(KopEvent event) {
+        try {
+            putLock.lock();
+            queue.clear();
+            put(event);
+        } finally {
+            putLock.unlock();
+        }
+    }
+
+    static class KopEventThread extends ShutdownableThread {
+
+        public KopEventThread(String name) {
+            super(name);
+        }
+
+        @Override
+        protected void doWork() {
+            KopEvent event = null;
+            try {
+                event = queue.take();
+                event.process();
+            } catch (InterruptedException e) {
+                log.error("Error processing event {}, {}", event, e);
+            }
+        }
+
+    }
+
+    private void registerChildChangeHandler() {
+        metadataStore.registerListener(this::handleChildChangePathNotification);
+
+        // Really register ChildChange notification.
+        metadataStore.getChildren(getDeleteTopicsPath());
+        // init local kop brokers cache
+        getBrokers(metadataStore.getChildren(getBrokersChangePath()).join());
+    }
+
+    private void handleChildChangePathNotification(Notification notification) {
+        if (notification.getPath().equals(LoadManager.LOADBALANCE_BROKERS_ROOT)) {
+            this.brokersChangeHandler.handleChildChange();
+        } else if (notification.getPath().equals(getDeleteTopicsPath())) {
+            this.deletionTopicsHandler.handleChildChange();
+        }
+    }
+
+    private void getBrokers(List<String> pulsarBrokers) {
+        HashSet<Node> kopBrokers = Sets.newHashSet();
+        pulsarBrokers.forEach(broker -> {
+            try {
+                Optional<GetResult> brokerData = metadataStore.get(
+                        getBrokersChangePath() + "/" + broker).join();
```

Comment:
`join()` throws unchecked exception, which means you don't want to catch this exception. If you want to handle the exceptional case, use `get()` and don't catch `Exception`, catch the checked exceptions.

But I also concerns the synchronous call here. Because I found `getBrokers` could also be called in `BrokersChangeEvent#process`, which is a callback. Can it be changed to asynchronous call?